### PR TITLE
Add ninja_syntax recipe

### PR DESCRIPTION
**Pull request by dfroger**

### CODE REVIEW

**Comment by dfroger** [comment]
Note: My goal is to package the [craftr](https://github.com/craftr-build/craftr) build system, this is one of its depenencies.

File: recipes/ninja_syntax/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  run:
+    - python
+
+test:
+  imports:
+    - ninja_syntax
+
+about:
+  home: https://pypi.python.org/pypi/ninja_syntax/1.6.0
+  license: Apache-2.0
```
Comment (scopatz) [inline recipes/ninja_syntax/meta.yaml]: Can you include the `license_file`, please? It is required for Apache 2

Comment (dfroger) [inline recipes/ninja_syntax/meta.yaml]: Ok, license file is not in the source archive, but in the repo.

I'll run into the same issue as [git-lfs recipe](https://github.com/conda-forge/staged-recipes/pull/2015).

Comment (dfroger) [inline recipes/ninja_syntax/meta.yaml]: (which is now fixed)

**Comment by dfroger** [comment]
The license file is added!

File: recipes/ninja_syntax/COPYING [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (scopatz) [inline recipes/ninja_syntax/COPYING]: This has to have the actual copyright of the package

Comment (dfroger) [inline recipes/ninja_syntax/COPYING]: I opened a [ninja issue](https://github.com/ninja-build/ninja/issues/1212)

File: recipes/ninja_syntax/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{% set version = "1.6.0" %}
+{% set sha256 = "33c203b4ee39a638c9af757ed1c4257e05abf1fae0615305d103020f36c804c5" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/43/d2/7a9cafe2b3b6ee260dfeee1a963630c8f0c16620c0ab7d18caab7fb46a42/{{ name }}-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/ninja_syntax/meta.yaml]: Can you please copy and paste the URL from the example here?

File: recipes/ninja_syntax/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install
```
Comment (jakirkham) [inline recipes/ninja_syntax/meta.yaml]: Needs `--single-version-externally-managed --record record.txt` if it uses `setuptools`.

File: recipes/ninja_syntax/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    - setuptools
+  run:
+    - python
+
+test:
+  imports:
+    - ninja_syntax
+
+about:
+  home: https://pypi.python.org/pypi/ninja_syntax/1.6.0
```
Comment (jakirkham) [inline recipes/ninja_syntax/meta.yaml]: Looks like the homepage is https://github.com/ninja-build/ninja . If so, is this a dup of [`ninja`]( https://github.com/conda-forge/ninja-feedstock ), which already have?

File: recipes/ninja_syntax/COPYING (hunk 2 above)
Comment (dfroger) [inline recipes/ninja_syntax/COPYING]: This is now fixed

File: recipes/ninja_syntax/meta.yaml (hunk 3 above)
Comment (dfroger) [inline recipes/ninja_syntax/meta.yaml]: done!

File: recipes/ninja_syntax/meta.yaml (hunk 4 above)
Comment (dfroger) [inline recipes/ninja_syntax/meta.yaml]: It's using `distutils`...

File: recipes/ninja_syntax/meta.yaml (hunk 5 above)
Comment (dfroger) [inline recipes/ninja_syntax/meta.yaml]: The [ninja_syntax package on pypi](https://pypi.python.org/pypi/ninja_syntax/1.6.0) provides just one file, https://github.com/ninja-build/ninja/blob/master/misc/ninja_syntax.py .

So it's a package for on file of https://github.com/ninja-build/ninja. This file will be duplicated in packages `ninja` and `ninja_syntax`.

`ninja_syntax` allows python modules to generate .ninja files, without installing the entire `ninja` (which imply compiling C++).

For a conda point of view, one package would be enough, but software such as [craftr](https://github.com/craftr-build/craftr) do [import](https://github.com/craftr-build/craftr/blob/cc94a5440eb3d7b31ab8e5f516a74880f040a309/craftr/core/build.py#L32) `ninja_syntax`.

**Comment by dfroger** [comment]
All issues shoud be now fixed

**Comment by scopatz** [comment]
Works for me on local docker on linux.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**